docs/examples/plot_partial_module_shading_simple.py

```diff
@@ -0,0 +1,273 @@
+    sde_args = pvsystem.calcparams_desoto(
+        Geff,
+        Tcell,
+        alpha_sc=parameters['alpha_sc'],
+        a_ref=parameters['a_ref'],
+        I_L_ref=parameters['I_L_ref'],
+        I_o_ref=parameters['I_o_ref'],
+        R_sh_ref=parameters['R_sh_ref'],
+        R_s=parameters['R_s'],
+    )
+    # sde_args has values:
+    # (photocurrent, saturation_current, resistance_series,
+    # resistance_shunt, nNsVth)
+
+    # Use Bishop's method to calculate points on the IV curve with V ranging
+    # from the reverse breakdown voltage to open circuit
+    kwargs = {
+        'breakdown_factor': parameters['breakdown_factor'],
+        'breakdown_exp': parameters['breakdown_exp'],
+        'breakdown_voltage': parameters['breakdown_voltage'],
+    }
+    v_oc = singlediode.bishop88_v_from_i(
+        0.0, *sde_args, method=method, **kwargs
+    )
+    vd = np.linspace(0.99*kwargs['breakdown_voltage'], v_oc, ivcurve_pnts)
+
+    ivcurve_i, ivcurve_v, _ = singlediode.bishop88(vd, *sde_args, **kwargs)
+    return pd.DataFrame({
+        'i': ivcurve_i,
+        'v': ivcurve_v,
+    })
+
+
+def plot_curves(dfs, labels):
+    """plot the forward- and reverse-bias portions of an IV curve"""
+    fig, axes = plt.subplots(1, 2)
+    for df, label in zip(dfs, labels):
+        df.plot('v', 'i', label=label, ax=axes[0])
+        df.plot('v', 'i', label=label, ax=axes[1])
+        axes[0].set_xlim(right=0)
+        axes[1].set_xlim(left=0)
+
+    return axes
+
+
+cell_curve_full_sun = simulate_full_curve(cell_parameters, Geff=1000, Tcell=40)
+cell_curve_shaded = simulate_full_curve(cell_parameters, Geff=200, Tcell=40)
+plot_curves([cell_curve_full_sun, cell_curve_shaded], ['Full Sun', 'Shaded'])
+plt.gcf().suptitle('Cell-level reverse- and forward-biased IV curves')
+
+
+# %%
+# Combining cell IV curves to create a module IV curve
+# ----------------------------------------------------
+#
+# To combine the individual cell IV curves and form a module's IV curve,
+# the cells in each substring must be added in series and the substrings
+# added in parallel.  To add in series, the voltages for a given current are
+# added.  However, because each cell's curve is discretized and the currents
+# might not line up, we align each curve to a common set of current values
+# with interpolation.
+
+
+def interpolate(df, i):
+    """convenience wrapper around scipy.interpolate.interp1d"""
+    f_interp = interp1d(np.flipud(df['i']), np.flipud(df['v']), kind='linear',
+                        fill_value='extrapolate')
+    return f_interp(i)
+
+
+def combine_series(dfs):
+    """
+    Combine IV curves in series by aligning currents and summing voltages.
+    The current range is based on the first curve's current range.
+    """
+    df1 = dfs[0]
+    imin = df1['i'].min()
+    imax = df1['i'].max()
+    i = np.linspace(imin, imax, 1000)
+    v = 0
+    for df2 in dfs:
+        v_cell = interpolate(df2, i)
+        v += v_cell
```

Comment:
Or you could put the bypass diode here using `v = v.clip(-bypass_voltage)` like [Cliff commented](#r427429840) _except note the **negative** sign_, where `bypass_voltage` would be an argument of `combine_series(..., bypass_voltage=0.5)` and 0.5[V] is a typical diode trigger voltage (_aka_ [forward voltage drop or _threshold_ voltage](https://en.wikipedia.org/wiki/Diode#Forward_bias)). 

From this it's clearer that at the substring/submodule level reverse bias is not important because the substring is either bypassed or not. And the entire substring tends to behave near Vmp like the most shaded cell in the substring.

The reverse bias part of each cell is really only useful to tell us where the substring shifts from the most shaded cell to the rest of the substring, where there's a bump/jump in the current. This inflection or discontinuity occurs where the shaded cells drop into reverse bias because they can't handle the current, _ie_ after the first knee in the IV curve (usu. near Vmp, but not necessarily), the voltage of the shaded cells suddenly drops from positive to negative until the break down voltage, where the second knee starts, but unfortunately if the total breakdown voltage of the shaded cells _plus_ the forward bias of the remaining cells is less than the negative of the bypass diode trigger voltage, `-Vbypass`, then the substring is bypassed, and the voltage doesn't go any lower than `-Vbypass`.

Reply:
Could add the bypass diode here, but I added it where @cwhanse suggested so that it is easier to modify this example to use his forthcoming `add_series` function (I assume his function won't consider bypass diodes).